app.rb

```diff
@@ -3,27 +3,51 @@
 require 'yaml'
 require 'berbix'
 require 'namey'
+require 'json'
+require 'sinatra/custom_logger'
+require 'logger'
+
+set :logger, Logger.new(STDOUT)
 
 berbix_config = YAML.load(File.read('berbix_config.yaml'))
 
 get '/' do
   @generator = Namey::Generator.new
   @username = @generator.name(:common)
-  @uid = rand(10000000000)
+  
+  if params[:uid]
+    @uid = params[:uid]
+  else
+    @uid = 123
+  end
 
   client = Berbix::Client.new(
     client_secret: berbix_config['client_secret'],
     environment: :production
   )
 
-  @transaction_tokens = client.create_transaction(
-    customer_uid: @uid, # ID for the user in client database
-    template_key: berbix_config['template_key'], # Template key for this transaction
-  )
-
-  # To avoid dealing with a DB for this demo, we'll write the refresh token to a file
-  File.open("refresh_token.txt", "w") do |file|
-    file.write(@transaction_tokens.refresh_token)
+  # Check if refresh token exists for given UID
+  if File.file?("#{@uid}_refresh_token.txt")
+    # read the refresh token from disk
+    @refresh_token = File.open("#{@uid}_refresh_token.txt").read
+    
+    begin
+      @transaction_tokens = Berbix::Tokens.from_refresh(@refresh_token)
+      @data = client.fetch_transaction(@transaction_tokens) # needed to force library to refresh the refresh token
```

Comment:
I'm mainly concerned to see if the above 2 lines are correct and recommended.

Reply:
You can call `refresh_tokens` to get a new set of fresh tokens if your goal is to get a new `client_token` - I'm guessing that's what you meant, right? You don't need to fetch a new `refresh_token` given that it's long-lived.

Reply:
Ah I tried that here and I got this error:
<img width="523" alt="Screen Shot 2020-11-16 at 9 28 03 PM" src="https://user-images.githubusercontent.com/31038/99342997-ac6a2980-2852-11eb-8465-df032e3e514a.png">

I'll try to debug tomorrow.